src/PactNet.AspNetCore/ProviderState/ProviderStateMiddleware.cs

```diff
@@ -0,0 +1,133 @@
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using PactNet.Verifier.ProviderState;
+
+namespace PactNet.AspNetCore.ProviderState
+{
+    /// <summary>
+    /// Defines the provider state middleware
+    /// It provides an endpoint to trigger the provider states before verification of pact interaction
+    /// </summary>
+    public class ProviderStateMiddleware
+    {
+        /// <summary>
+        /// The next pointer
+        /// </summary>
+        private readonly RequestDelegate next;
+        
+        /// <summary>
+        /// The provider state middleware options
+        /// </summary>
+        private readonly ProviderStateOptions options;
+
+        /// <summary>
+        /// The provider state accessor
+        /// </summary>
+        private readonly IProviderStateAccessor providerStateAccessor;
+
+        ///// <summary>
+        ///// the after interaction callback
+        ///// </summary>
+        //private Dictionary<string, Action> afterInteractionCallbacks;
```

Comment:
Not sure what this commented out code is for - does it need removing or is it unfinished?

Reply:
haha, leftover code